packages/devtools_app/lib/src/timeline/timeline_model.dart

```diff
@@ -256,26 +249,39 @@ class FullTimelineEventGroup {
   }
 
   void _addEventAtDisplayRow(TimelineEvent event, {@required int row}) {
-    if (row + event.displayDepth >= eventsByRow.length) {
-      for (int i = eventsByRow.length; i < row + event.displayDepth; i++) {
-        eventsByRow.add([]);
-        lastEventsByRow.add(null);
+    if (row + event.displayDepth >= rows.length) {
+      for (int i = rows.length; i < row + event.displayDepth; i++) {
+        rows.add(FullTimelineRowData());
       }
     }
 
     for (int i = 0; i < event.displayDepth; i++) {
       final displayRow = event.displayRows[i];
       for (var e in displayRow) {
-        eventsByRow[row + i].add(e);
+        rows[row + i].events.add(e);
         if (e.time.end >
-            (lastEventsByRow[row + i]?.time?.end ?? const Duration())) {
-          lastEventsByRow[row + i] = e;
+            (rows[row + i].lastEvent?.time?.end ?? const Duration())) {
+          rows[row + i].lastEvent = e;
         }
       }
     }
   }
 }
 
+class FullTimelineRowData {
+  /// Timeline events that will be painted in this row in a visualization of a
```

Comment:
tiny nit: that will be painted --> "displayed"

Reply:
done